src/compiler/sexp.rs

```diff
@@ -219,7 +219,7 @@ enum SExpParseState {
 }
 
 #[derive(Debug)]
-enum SExpParseResult {
+enum SExpParseResult {  // the result of a call to parse an SExp
```

Comment:
seems useful to add that this allows the parser to be layered so lists can contain the same parser.